console-subscriber/src/lib.rs

```diff
@@ -343,11 +351,32 @@ impl proto::tasks::tasks_server::Tasks for Server {
             tonic::Status::internal("cannot start new watch, aggregation task is not running")
         })?;
         let (tx, rx) = mpsc::channel(self.client_buffer);
-        permit.send(Watch(tx));
+        permit.send(WatchKind::TaskUpdate(Watch(tx)));
         tracing::debug!("watch started");
         let stream = tokio_stream::wrappers::ReceiverStream::new(rx);
         Ok(tonic::Response::new(stream))
     }
+
+    async fn watch_task_details(
+        &self,
+        req: tonic::Request<proto::tasks::DetailsRequest>,
+    ) -> Result<tonic::Response<Self::WatchTaskDetailsStream>, tonic::Status> {
+        let task_id = req
+            .into_inner()
+            .id
+            .ok_or_else(|| tonic::Status::invalid_argument("missing task_id"))?;
+        let permit = self.subscribe.reserve().await.map_err(|_| {
+            tonic::Status::internal("cannot start new watch, aggregation task is not running")
+        })?;
+        let (tx, rx) = mpsc::channel(self.client_buffer);
+        permit.send(WatchKind::TaskDetailUpdate(task_id, Watch(tx)));
+        tracing::debug!("task details watch started");
+        // At this point we don't know if the task with the given ID exists or not.
+        // But the only Watch sender will be dropped if the task doesn't exist
+        // so that should close the stream.
```

Comment:
Pushed https://github.com/tokio-rs/console/pull/47/commits/fc982d25e2fa25937ca79c83f368a5909bd85e24 to solve this. I believe this is slightly different from what you suggested, so please let me know if you like this. I wanted to avoid sending a separate message before the actual `permit.send(WatchKind)` call, in order to avoid performing error handling twice.